### PR TITLE
Implement cache refresh after file changed

### PR DESCRIPTION
**Pull request by beinan**
fix issue https://github.com/prestodb/presto/issues/22019

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java [hunk 1]
```diff
@@ -31,30 +31,30 @@ public ExtendedFileSystem createCachingFileSystem(
...
-            boolean validationEnabled)
+            CacheConfig cacheConfig)
             throws IOException
     {
         if (!cachingEnabled) {
             return fileSystem;
         }
 
-        checkState(cacheType != null);
+        checkState(cacheConfig.getCacheType() != null);
```
Comment (ChunxuTang) [inline presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java:41]: nit: Do you also want to check whether `cacheConfig` is null or not?

**Review by ChunxuTang — APPROVED** [review]
LGTM.
@beinan Thanks for the fix!

**Comment by ChunxuTang** [comment]
Re-running the failed test.

**Comment by nmahadevuni** [comment]
@beinan Why don't we do this by default? So, we do not encounter such staleless problems at all.

File: presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java (hunk 1 above)
Comment (beinan) [reply presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java]: good catch,  just fixed it

**Comment by beinan** [comment]
@zacw7 do you also want to take a look at this PR?  The change is behind a newly added flag which is disabled by default, so I think there should be no impact on Meta's production.

**Review by ChunxuTang — APPROVED** [review]

**Review by zacw7 — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by beinan** [comment]
force push to rerun the test

**Pull request merged**